lib/index.js

```diff
@@ -828,8 +828,13 @@ Stream.prototype.destroy = function () {
     _(this._consumers).each(function (consumer) {
         self._removeConsumer(consumer);
     });
+    _(this._observers).each(function (observer) {
+        self._removeObserver(observer)
+    });
+
     if (this.source) {
         this.source._removeConsumer(this);
+        this.source._removeObserver(this);
```

Comment:
Actually, I think that means you don't need the `_removeObserver` method at all. Just set `this._observers` to `null` (or empty array).